src/ad-cue-tags.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file ad-cue-tags.js
+ */
+import window from 'global/window';
+
+/**
+ * Searches for an ad cue that overlaps with the given mediaTime
+ */
+const findAdCue = function(track, mediaTime) {
+  let cues = track.cues;
+
+  for (let i = 0; i < cues.length; i++) {
+    let cue = cues[i];
+
+    if (mediaTime >= cue.adStartTime && mediaTime <= cue.adEndTime) {
+      return cue;
+    }
+  }
+  return null;
+};
+
+const initAdCueTrack = function(controller) {
```

Comment:
It's a bit odd to pass the caller to this function so that it can set a value on the caller. If we want to separate responsibility, we can either create an AdCueTagManager (or something similar) that maintains its own state (keeping hold of the cueTagsTrack) with MasterPlaylistController having only an instance of the manager, or create the cue tags track in MasterPlaylistController (move the innards of this function to MasterPlaylistController) and continue to pass in that reference as you do later on. I think the latter might make the most sense, as these functions are merely utility functions, and MasterPlaylistController is the one who should know a bit more about its state, and what gets added to Video.js type APIs (i.e., the new text track).